crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,790 @@
+                            debug!("create drive file_id:{},path:{}", file_id, path);
+                            backend.file_map.insert(file_id, file);
+                            backend.file_path_map.insert(file_id, path.clone());
+                            return make_create_drive_resp(
+                                req_inner.device_io_request,
+                                req_inner.create_disposition,
+                                file_id,
+                            );
+                        }
+                        Err(error) => {
+                            warn!(%error, "Open file dir error");
+                            //return by downside
+                        }
+                    }
+                }
+                //create disposition is not correct
+                let io_response = DeviceIoResponse::new(req_inner.device_io_request, NtStatus::UNSUCCESSFUL);
+                let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+                    device_io_reply: io_response,
+                    file_id,
+                    information: Information::empty(),
+                });
+                return Ok(vec![SvcMessage::from(res)]);
+            }
+        }
+    }
+
+    let mut fs = std::fs::OpenOptions::new();
+    if CreateDisposition::FILE_OPEN_IF == req_inner.create_disposition {
+        fs.create(true).write(true).read(true);
+    }
+    if CreateDisposition::FILE_CREATE == req_inner.create_disposition {
+        fs.create_new(true).write(true).read(true);
+    }
+    if CreateDisposition::FILE_SUPERSEDE == req_inner.create_disposition {
+        fs.create(true).write(true).append(true).read(true);
+    }
+    if CreateDisposition::FILE_OPEN == req_inner.create_disposition {
+        fs.read(true);
+    }
+    if CreateDisposition::FILE_OVERWRITE == req_inner.create_disposition {
+        fs.write(true).truncate(true).read(true);
+    }
+    if CreateDisposition::FILE_OVERWRITE_IF == req_inner.create_disposition {
+        fs.write(true).truncate(true).create(true).read(true);
+    }
+
+    match fs.open(&path) {
+        Ok(file) => {
+            debug!("create drive file_id:{},path:{}", file_id, path);
+            backend.file_map.insert(file_id, file);
+            backend.file_path_map.insert(file_id, path.clone());
+            make_create_drive_resp(req_inner.device_io_request, req_inner.create_disposition, file_id)
+        }
+        Err(error) => {
+            warn!(?error, "Open file error for path:{}", path);
+            let io_response = DeviceIoResponse::new(req_inner.device_io_request, NtStatus::UNSUCCESSFUL);
+            let res = RdpdrPdu::DeviceCreateResponse(DeviceCreateResponse {
+                device_io_reply: io_response,
+                file_id,
+                information: Information::empty(),
+            });
+            Ok(vec![SvcMessage::from(res)])
+        }
+    }
+}
+pub(crate) fn process_dependent_file(
```

Comment:
style: Use some newlines to separate functions/types definitions.
```suggestion
}

pub(crate) fn process_dependent_file(
```